### PR TITLE
Add API for adding users

### PR DESCRIPTION
**Pull request by skaex**
### Description
This PR adds an API for adding users to join.

### Motivation and Context
- I need an API for platform to be able to create users on join server

https://app.asana.com/0/1179330853755533/1189555546176041

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### How to test
- Make a http post request to `/api/users` with the user payload

### Checklist:

- [x] My code follows the code style of this project.
- [x] I have added neccessary labels
- [x] My PR title follows the convention in the guide
- [x] I have assigned myself to the PR
- [x] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [x] All new and existing tests passed.

### CODE REVIEW

File: env.sample [hunk 1]
```diff
@@ -160,3 +160,4 @@ MAILCHIMP_LIST_ID=123
 # This is a password we create that just needs to match here and on the cloudmailin target
 # it can be anything at all and doesn't actually serve a very important role; just a minor filter.
 CLOUDMAILIN_PASSWORD=random_password
+API_CONN_KEY=random_key
```
Comment (sadleb) [inline env.sample:163]: Can we call this `JOIN_API_TOKEN` and put a comment here about the fact that it’s used by other servers to access the api? Once we add matching ENV vars on platform (or other servers), I think it’ll be less confusing to call them all the same thing and we would call it `API_CONN_KEY` in platform code

**Review by sadleb — CHANGES_REQUESTED** [review]

File: app/models/user.rb [hunk 2]
```diff
@@ -230,6 +230,11 @@ def record_converted_on_salesforce(contact_id)
     self.create_mailchimp
   end
 
+  # I don't want to use serializers. It'll be overkill for now
+  def as_json(opts = {})
```
[the remark this answers is not in the document]
Comment (skaex) [reply app/models/user.rb:234]: Yeah it is.

File: app/controllers/api/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Api::UsersController < ApiController
+  def index
+    @users = User.where(email: params[:q])
+
+    json_response @users
+  end
+
+  def create
+    @user = User.new(user_params)
```
[the remark this answers is not in the document]
Comment (skaex) [reply app/controllers/api/users_controller.rb:9]: I used `.save!` below. It'll throw an error on validation errors.

[the remark this answers is not in the document]
Comment (skaex) [reply app/controllers/api/users_controller.rb:9]: The constraint on email comes from [devise's](https://github.com/heartcombo/devise/blob/master/lib/devise/models/validatable.rb) `validatable`

**Review by sadleb — APPROVED** [review]
lgtm. 

Note to other readers, we're being more lax since we're trying to get rid of the Join server (aka this repo). This is just a stop-gap to not have to cut-over the entire Braven Network feature to the platform code at this moment.

**Pull request merged**